lua/entities/gmod_tardis/modules/sh_spin.lua

```diff
@@ -24,52 +25,54 @@ end
 
 if SERVER then
     ENT:AddHook("Initialize", "spin", function(self)
-        self:SetData("spindir", -1, true)
-        self:SetData("spindir_prev", 0, true)
+        self:SetData("spin", true, true)
+        self:SetData("spin-dir", -1, true)
     end)
 
     function ENT:SetSpinDir(dir)
-        self:CallHook("SpinChanged", dir)
-        return self:SetData("spindir", dir, true)
+        self:SetData("spin-dir", dir, true)
+        self:CallHook("SpinChanged", self:GetSpinDir())
     end
 
-    function ENT:ToggleSpin()
-        local current = self:GetData("spindir", -1)
-        local prev = self:GetData("spindir_prev", 0)
+    function ENT:SetSpin(on, dir)
+        self:SetData("spin", on, true)
+        if on and dir ~= nil then
```

Comment:
Perhaps we should allow changing dir even if spin if off?

Just like we do here:
![image](https://github.com/MattJeanes/TARDIS/assets/32988384/52dd429a-e6e2-4dd3-8c6f-5280b34acc91)

Reply:
Don't see why not I suppose

https://github.com/MattJeanes/TARDIS/pull/969/commits/88d627855679156562a989a54535a3815630e1b1